lifetimes/fitters/pareto_nbd_fitter.py

```diff
@@ -147,9 +150,19 @@ def conditional_probability_alive(self, frequency, recency, T):
         x, t_x = frequency, recency
         r, alpha, s, beta = self._unload_params('r', 'alpha', 's', 'beta')
 
-        A_0 = np.exp(self._log_A_0([r, alpha, s, beta], x, t_x, T))
-        return 1. / (1. + (s / (r + s + x)) *
-                     (alpha + T) ** (r + x) * (beta + T) ** s * A_0)
+        maxab = max(alpha, beta)
+        absab = abs(alpha - beta)
+        if alpha < beta:
+            param2 = r + x
+        else:
+            param2 = s + 1.
+
+        F0 = ((alpha + T)**(r + x)) * ((beta + T)**s)
+        F1 = hyp2f1(r + s + x, param2, r + s + x + 1., absab / (maxab + t_x)) / \
+            ((maxab + t_x)**(r + s + x))
+        F2 = hyp2f1(r + s + x, param2, r + s + x + 1., absab / (maxab + t_x)) / \
+            ((maxab + T)**(r + s + x))
```

Comment:
@CamDavidsonPilon the best solution I can see after today's effort is a check that protects against `(recency + frequency) > some_constant` (where `some_constant` is about 300 or so). The problem is that `(maxab + T)` is a `float` and when raised to a very large exponent floating point arithmetic fails. Some cursory investigation of `numpy` and `mpmath` didn't yield a solution.

Reply:
posted on the other page and reposting it here for consistency:

guess this is easier to write here, instead of pull request.

I think instead of trying the computational approach, if we use the alternative and more simplified expression for P_Alive which is Eq. 34 in the reference: http://brucehardie.com/notes/009/pareto_nbd_derivations_2005-11-05.pdf, this problem will be fixed.

I tried it , and it eliminates the overflow for high values of R,F, and T (verified for range of values >300).

below is the function with expression in 34. Note that parts of this equation as indeed present in conditional expected number of purchases.

```
def conditional_probability_alive_2(model, frequency, recency, T):
        params=model.params_.values()
        r,alpha,s,beta= params
        x, t_x = frequency, recency
        likelihood = -model._negative_log_likelihood(params, x, t_x, T,0.0) 
        first_term = gammaln(r + x) - gammaln(r) + r * log(alpha) + s * \
            log(beta) - (r + x) * log(alpha + T) - s * log(beta + T) 
        return exp(first_term - likelihood)
```



Thank you very much for the package.
Aravind.